app/components/channel_drawer/channels_list/list/index.js

```diff
@@ -26,9 +27,10 @@ function mapStateToProps(state) {
     const publicChannelIds = getSortedPublicChannelIds(state);
     const privateChannelIds = getSortedPrivateChannelIds(state);
     const directChannelIds = getSortedDirectChannelIds(state);
+    const currentTeamId = getCurrentTeamId(state);
 
     return {
-        canCreatePrivateChannels: showCreateOption(config, license, General.PRIVATE_CHANNEL, isAdmin(roles), isSystemAdmin(roles)),
+        canCreatePrivateChannels: showCreateOption(state, config, license, currentTeamId, General.PRIVATE_CHANNEL, isAdmin(roles), isSystemAdmin(roles)),
```

Comment:
I think we should use selectors for things like this in the future since explicitly passing in all these other fields each time is a lot of work, but I don't think the existing ones need to be changed until we start adding other permission checks

Reply:
Yes, I want to move that to selectors, but there are a cyclic dependency problem, and I prefer to defer it, and try to refactor the cyclic dependency in other User Story.